src/stream/src/executor/backfill/utils.rs

```diff
@@ -799,6 +805,30 @@ pub(crate) async fn persist_state<S: StateStore, const IS_REPLICATED: bool>(
     Ok(())
 }
 
+pub type BackfillRateLimiter =
+    RateLimiter<NotKeyed, InMemoryState, MonotonicClock, NoOpMiddleware<Instant>>;
+
+/// Creates a data chunk builder for snapshot read.
+/// If the `rate_limit` is smaller than `chunk_size`, it will take precedence.
+/// This is so we can partition snapshot read into smaller chunks than chunk size.
+pub fn create_builder_and_limiter(
+    rate_limit: Option<usize>,
+    chunk_size: usize,
+    data_types: Vec<DataType>,
+) -> (DataChunkBuilder, Option<BackfillRateLimiter>) {
+    if let Some(rate_limit) = rate_limit
+        && rate_limit > 0
+        && rate_limit < chunk_size
+    {
+        (
+            DataChunkBuilder::new(data_types, rate_limit),
+            Some(create_limiter(rate_limit)),
+        )
+    } else {
+        (DataChunkBuilder::new(data_types, chunk_size), None)
```

Comment:
I didn't get it. So the `rate_limit` will not take effect if it's out of the range `(0, chunk_size)`?

Reply:
fix